source/AI.cpp

```diff
@@ -3686,17 +3687,27 @@ void AI::IssueOrders(const PlayerInfo &player, const Orders &newOrders, const st
 	// toggle is a move command; it always counts as a new command.
 	bool hasMismatch = isMoveOrder;
 	bool gaveOrder = false;
+	
+	// A target is valid if we have no target, or when the target is in the
+	// same system as the flagship.
+	bool isValidTarget = !newTarget || (newTarget && player.Flagship() &&
+		newTarget->GetSystem() == player.Flagship()->GetSystem());
+	
 	for(const Ship *ship : ships)
 	{
 		// Never issue orders to a ship to target itself.
 		if(ship == newTarget)
 			continue;
 		
-		// Never issue orders that target a ship in another system.
-		if(newTarget && ship->GetSystem() != newTarget->GetSystem())
-			continue;
-		
 		gaveOrder = true;
+		
+		// If we don't have a valid target, then we cancel the orders for
```

Comment:
Refactor the loop to avoid this. Probably we can do this with a `find_if` lambda.

Reply:
Refactored this by taking it out of the loop.